DeviceAdministration/Web/WebApiControllers/TelemetryApiController.cs

```diff
@@ -335,8 +333,49 @@ public class TelemetryApiController : WebApiControllerBase
                 {
                     try
                     {
-                        deviceId = DeviceSchemaHelper.GetDeviceID(devInfo);
-                        props = DeviceSchemaHelper.GetDeviceProperties(devInfo);
+                        deviceId = devInfo.DeviceProperties.DeviceID;
+                        props = devInfo.DeviceProperties;
+                        enabledState = props.HubEnabledState;
+                    }
+                    catch (DeviceRequiredPropertyNotFoundException)
+                    {
+                        continue;
+                    }
```

Comment:
But now this exceptions won't be thrown, nullargumentexception would be thrown or something similar in C#